lib/fs.js

```diff
@@ -2712,9 +2712,8 @@ realpath.native = (path, options, callback) => {
 function mkdtemp(prefix, options, callback) {
   callback = makeCallback(typeof options === 'function' ? options : callback);
   options = getOptions(options, {});
-  if (!prefix || typeof prefix !== 'string') {
-    throw new ERR_INVALID_ARG_TYPE('prefix', 'string', prefix);
-  }
+
+  validateString(prefix, 'prefix');
```

Comment:
I'm not sure if it matters or not, but this will accept only a string while the existing code accepts `undefined`, `null`, and other falsy values.

Reply:
I guess that's handled by `nullCheck()`...

Reply:
The test change shows that empty strings are now allowed, though. Is that okay?

Reply:
Currently the error throw when passing the empty string is quite confusing to be honest: `TypeError [ERR_INVALID_ARG_TYPE]: The "prefix" argument must be of type string. Received type string ('')`.

Reply:
I agree it's confusing, but does the function "work" with an empty string? what does it do?

Reply:
Yes, I think `mkdtemp()` does work with an empty prefix string. The only requirement is that the last 6 characters of the template must be `XXXXXX`. It just replaces the 6 trailing X characters (or more if present on BSD) of the template with random characters to make the directory name unique and makes the directory.

cc @jasnell Since this was added in https://github.com/nodejs/node/commit/060e5f0c0064e578c2150f13e3f91ac15fdeed92.